src/model/article-information.ts

```diff
@@ -67,6 +87,152 @@ export class ArticleInformation extends BackmatterEntity {
+            });
+
+            return subjectGroupXml;
+        }
+    }
+
+    private createDoiXml(xmlDoc: Document) {
+        if (this.articleDOI) {
+            const doiXml = xmlDoc.createElement('article-id');
+            doiXml.setAttribute('pub-id-type', 'doi');
+            doiXml.appendChild(xmlDoc.createTextNode(this.articleDOI));
+            return doiXml;
+        }
+    }
+
+    private createPublisherXml(xmlDoc: Document) {
+        if (this.publisherId) {
+            const publisherIdXml = xmlDoc.createElement('article-id');
+            publisherIdXml.setAttribute('pub-id-type', 'publisher-id');
+            publisherIdXml.appendChild(xmlDoc.createTextNode(this.publisherId));
+            return publisherIdXml;
+        }
+    }
+
+    private createElocationIdXml(xmlDoc: Document) {
+        if (this.elocationId) {
+            const elocationIdXml = xmlDoc.createElement('elocation-id');
+            elocationIdXml.appendChild(xmlDoc.createTextNode(this.elocationId));
+            return elocationIdXml;
+        }
+    }
+
+    private createVolumeXml(xmlDoc: Document) {
+        if (this.volume) { 
+            const volumeXml = xmlDoc.createElement('volume');
+            volumeXml.appendChild(xmlDoc.createTextNode(this.volume));
+            return volumeXml;
+        }
+    }
+
+    private createPublicationDateXml(xmlDoc: Document) {
+        const [year, month, day] = this.publicationDate.split('-');
+        const pubdateXml = xmlDoc.createElement('pub-date');
+        pubdateXml.setAttribute('date-type', 'pub');
+        pubdateXml.setAttribute('publication-format', 'electronic');
+        pubdateXml.setAttribute('iso-8601-date', this.publicationDate.replace(/-+$/, ''));
+
+        if (!year) {
+            return;
+        }
+
+        const yearXml = xmlDoc.createElement('year');
+        yearXml.appendChild(xmlDoc.createTextNode(year));
+        pubdateXml.appendChild(yearXml);
+
+        if (month) {
+            const monthXml = xmlDoc.createElement('month');
+            monthXml.appendChild(xmlDoc.createTextNode(month));
+            pubdateXml.appendChild(monthXml);
+        }
+
+        if (day) {
+            const dayXml = xmlDoc.createElement('day');
+            dayXml.appendChild(xmlDoc.createTextNode(day));
+            pubdateXml.appendChild(dayXml);
+        }
+
+        return pubdateXml;
+    }
+
+    private createLicenseXml(xmlDoc: Document) {
+        if (!this.licenseType) return;
+        const licenseXml = xmlDoc.createElement('license');
+        licenseXml.setAttribute('xlink:href', LICENSE_URL_MAP[this.licenseType]);
+        const licenseRef = xmlDoc.createElement('ali:license_ref');
+        licenseRef.appendChild(xmlDoc.createTextNode(LICENSE_URL_MAP[this.licenseType]));
+        licenseXml.appendChild(licenseRef);
+
+        if (this.licenseType) {
+            const licenseText = parseXML(
+                '<wrapperNode xmlns:xlink="http://www.w3.org/1999/xlink"><license-p>' +
```

Comment:
Just cross checked against [the spec](https://github.com/libero/editor/issues/22), I don't see a `wrapperNode` type there at all. Is this needed?

Reply:
This is a work around for a feature of the `xmldom` library. We need a `xmlns:xlink="http://www.w3.org/1999/xlink"` attribute somewhere in the xml otherwise the serializer adds an empty one for us which then causes us to have innvalid xml. Later on we pluck the `<license-p>` element from this document aand place it into the main document which has `xmlns:xlink="http://www.w3.org/1999/xlink"` in the `<article>` tag.